### DISCUSSION TITLE
How to add local ollama as a model

### DISCUSSION THREAD

**Discussion by thubreg** [mckaywrigley/chatbot-ui#1401] [Help]
I've got a virtual machine in the cloud with:
* Ollama up and running (confirmed with `curl http://localhost:11434/api/generate -d '{ "model": "llama2", "prompt":"Why is the sky blue?1 secntence only" }'` on the machine)
* Supabase up and running as a hosted service
* Chatbot-ui up and running (port open in firewall, can access the page at port 3000, am able to log in, click around, etc). Started chatbot-ui with NEXT_PUBLIC_OLLAMA_URL=http://localhost:11434 in the .env.local and using npm run chat

Now I go to models -> new model -> enter following parameters:
* Name: llama2
* Model ID: llama2
* Base url: I've tried http://localhost:11434, http://localhost:11434/api/generate, http://127.0.0.1:11434/api/generate, http://127.0.0.1:11434; all tested with curl and confirmed to be working
* API Key: empty

I see the request coming in in the backed, and in the browser I get "404 404 page not found". 

Question: How do I get Chatbot-ui to pick up on the local model?

P.S. the two warning I see in the backend are
* GenerateSW has been called multiple times, perhaps due to running webpack in --watch mode. The precache manifest generated after the first call may be inaccurate! Please see https://github.com/GoogleChrome/workbox/issues/1790 for more information.
* metadata.metadataBase is not set for resolving social open graph or twitter images, using "http://localhost:3000". See https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadatabase`

**Comment by Bortus-AI** [comment]
NEXT_PUBLIC_OLLAMA_URL has the be a public ip or domain since it's cloud hosted

**Reply by thubreg** [reply]
Ollama and Chatbot-UI are hosted on the same cloud machine.

**Reply by Bortus-AI** [reply]
Correct but if you're using a browser from another computer you have to use the public ip for ollama in the .env as it's passed to the browser

**Reply by Bortus-AI** [reply]
You also have to run OLLAMA_HOST=0.0.0.0 OLLAMA_ORIGINS=* ollama serve

**Reply by EduardDoronin** [reply]
I ran into a similar issue, where exactly do I have to run "OLLAMA_HOST=0.0.0.0 OLLAMA_ORIGINS=* ollama serve" @Bortus-AI?
